File: GandanBusJava/app/src/main/java/com/gandan/android/gandanbusjava/model/MsgBody.java
package com.gandan.android.gandanbusjava.model;

import org.simpleframework.xml.Element;
import org.simpleframework.xml.ElementList;

import java.util.ArrayList;
import java.util.List;


public class MsgBody {

    @ElementList(name = "busLocationList", inline = true)
    private ArrayList<BusLocationList> busLocationList = new ArrayList<>();

    public ArrayList<BusLocationList> getBusLocationList() {
        return busLocationList;
    }

    public void setBusLocationList(ArrayList<BusLocationList> busLocationList) {
        this.busLocationList = busLocationList;
    }
}

File: GandanBus/app/src/main/java/com/gandan/android/gandanbus/model/MsgBody.kt
package com.gandan.android.gandanbus.model

import org.simpleframework.xml.Element
import org.simpleframework.xml.ElementList
import org.simpleframework.xml.Root
import java.util.ArrayList

@Root (name = "response")
data class MsgBody(@field:ElementList(name = "busLocationList", type = BusLocationList::class, required = false, entry = "busLocationList", inline = true) var busLocationList: List<BusLocationList>) {
    constructor() : this(mutableListOf<BusLocationList>())
}